src/EditorFeatures/CSharpTest/Diagnostics/GenerateMethod/GenerateMethodTests.cs

```diff
@@ -2782,6 +2782,24 @@ private static object DeserializeObject(JToken token, Type type)
 }");
         }
 
+        [WorkItem(8010, "https://github.com/dotnet/roslyn/issues/8010")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateMethod)]
+        public async Task TestGenerateMethodFromStaticProperty()
+        {
+            await TestAsync(
+@"using System ; public class Test { public static int Property { get { return [|Method|] ( ) ; } } } ",
+@"using System ; public class Test { public static int Property { get { return Method ( ) ; } } private static int Method ( ) { throw new NotImplementedException ( ) ; } } ");
+        }
+
+        [WorkItem(8010, "https://github.com/dotnet/roslyn/issues/8010")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateMethod)]
+        public async Task TestGenerateMethodFromStaticProperty2()
+        {
+            await TestAsync(
+@"using System ; public class OtherClass { } public class Test { public static OtherClass Property { get { if ( s_field == null ) s_field = [|InitializeProperty|] ( ) ; return s_field ; } } private static OtherClass s_field ; } ",
+@"using System ; public class OtherClass { } public class Test { public static OtherClass Property { get { if ( s_field == null ) s_field = InitializeProperty ( ) ; return s_field ; } } private static OtherClass InitializeProperty ( ) { throw new NotImplementedException ( ) ; } private static OtherClass s_field ; } ");
```

Comment:
Yeah, this is really hard to understand. I was going to suggest changing from suffixing the test method names with numbers to using words to describe the difference, but then my head started hurting trying to understand nesting.